apps/files_sharing/lib/Controller/ShareesAPIController.php

```diff
@@ -197,6 +197,27 @@ public function search(string $search = '', ?string $itemType = null, int $page
 			$result['exact'] = array_merge($this->result['exact'], $result['exact']);
 		}
 		$this->result = array_merge($this->result, $result);
+
+		// Get a list of groups to exclude from search results
+		$blacklisted_groups = $this->config->getSystemValue('blacklisted_groups', true);
```

Comment:
```suggestion
		$blacklisted_groups = $this->config->getSystemValue('blacklisted_groups', true);
		if (!is_array($blacklisted_groups)) {
		    $blacklisted_groups = [];
		}
```

Graceful handling of non array is passed? This can be needless if the source if SURE to provide arrays (when probably not configurable)